test/unit/cmap/commands.test.js

```diff
@@ -0,0 +1,92 @@
+const { expect } = require('chai');
+const { Response } = require('../../../src/cmap/commands');
+
+describe('commands', function () {
+  describe('Response', function () {
+    describe('#parse', function () {
+      context('when the message body is invalid', function () {
+        const message = Buffer.from([]);
+        const header = {
+          length: 0,
+          requestId: 0,
+          responseTo: 0,
+          opCode: 0
+        };
+        const body = Buffer.from([]);
+
+        it('throws an exception', function () {
+          const response = new Response(message, header, body);
+          expect(() => response.parse()).to.throw();
```

Comment:
This throws because you can't read an int32 from no bytes, let's add to the throw condition to check for that (like message or error type)

Reply:
Now checks both for `RangeError` and the message.